executor/analyze.go

```diff
@@ -206,9 +213,62 @@ func (e *AnalyzeExec) Next(ctx context.Context, req *chunk.Chunk) error {
 			}
 		}
 	}
+	err = e.saveAnalyzeOptsV2()
+	if err != nil {
+		e.ctx.GetSessionVars().StmtCtx.AppendWarning(err)
+	}
 	return statsHandle.Update(e.ctx.GetInfoSchema().(infoschema.InfoSchema))
 }
 
+func (e *AnalyzeExec) saveAnalyzeOptsV2() error {
+	if !variable.PersistAnalyzeOptions.Load() || e.OptionsMap == nil {
+		return nil
+	}
+	for _, options := range e.OptionsMap {
+		err := saveAnalyzeOpts(options, e.ctx.(sqlexec.RestrictedSQLExecutor))
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
Better to wrap these updates into a single transaction, then if there is an error, it can rollback automatically and won't leave any incomplete data.

Reply:
A single update does not need a transaction right? Do you mean to wrap the updates of multiple tables or partitions in one transaction? Updates to part of the tables or partitions won't cause wrong functional behavior to analyze. But I agree it's a little confused to users because of the error while partial updated.

Reply:
Done